File: nutron-nutrition/src/main/java/com/lazysyntax/nutron/nutrition/model/entity/Meal.java
package com.lazysyntax.nutron.nutrition.model.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
import java.util.Objects;

@Entity
@Table(name = "meals")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Meal {
    @Id
    private String id;
    private String userId;
    private String name;
    private LocalDate date;

    @OneToMany(cascade = CascadeType.ALL, orphanRemoval = true)
    @JoinColumn(name = "meal_id") // Crea una clave foranea en la tabla "meal_food_snapshots"
    private List<MealFoodSnapshot> foods = new ArrayList<>();
}

File: nutron-app-kmp/composeApp/src/commonMain/kotlin/com/lazysyntax/nutron/domain/models/Meal.kt
package com.lazysyntax.nutron.domain.models

import kotlinx.datetime.LocalDate
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Meal(
    val id: String? = null,
    @SerialName("meal_name") val name: String,
    val foods: List<Food>? = null,
    val recipes: List<Recipe>? = null,
    val date: LocalDate? = null
)